riva/utils/opus/opus_client_decoder.h

```diff
@@ -74,34 +64,58 @@ class Decoder {
   [[nodiscard]] std::vector<std::vector<unsigned char>> DeserializeOpus(
       const std::vector<unsigned char>& opus) const;
 
+  /**
+   * Bitrate of decoded audio
+   * @return rate
+   */
+  [[nodiscard]] int Rate() const { return rate_; }
+
+  /**
+   * Number of channels of decoded audio
+   * @return channels
+   */
+  [[nodiscard]] int Channels() const { return channels_; }
+
+  /**
+   * Returns OpusTags structure after parsing, empty otherwise
+   * @return tags
+   */
+  [[nodiscard]] OpusTags Tags() const { return opus_tags_; }
+
+  /**
+   * Length of decoded audio in seconds
+   * @return length
+   */
+  [[nodiscard]] float Length() const { return length_; }
 
   template <typename T>
   static T ReadLittleEndian(const unsigned char* str)
   {
     T val = T();
     T shift = 0;
-    for (int i = 0; i < sizeof(T); ++i) {
+    for (int i = 0; i < (int)sizeof(T); ++i) {
       val += T(str[i] & 0xFF) << shift;
       shift += 8;
     }
     return val;
   }
 
+  /**
+ * If requested rate is not supported this helper computes nearest supported one.
+ * Returns 0 if no adjustment required.
+ * @param rate
+ * @return
+ */
+  static int32_t AdjustRateIfUnsupported(int32_t rate);
```

Comment:
Does this still return 0 if if no adjustment required?

Reply:
No, it doesn't